packages/cli/docs/custom-plugins.md

```diff
@@ -0,0 +1,702 @@
+# Integrating a custom plugin in the CLI
+
+One of the main features of Code PushUp is the ability to write custom plugins and track your own metrics.
+It enables you to implement nearly any kind of metric you want to track progress with minimum effort.
+In this section we will go through the steps needed to create a custom plugin and integrate it in your project.
+
+## Setup the core config
```

Comment:
Common mistake, but "setup" is a noun, the verb is "set up":

```suggestion
## Set up the core config
```

E.g.:
- _We have a great **setup** on this project._
- _I **set up** a CI workflow yesterday._